src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java

```diff
@@ -0,0 +1,370 @@
+package com.nike.cerberus.cli;
+
+import com.nike.cerberus.command.StackDelegate;
+import com.nike.cerberus.command.cms.CreateCmsClusterCommand;
+import com.nike.cerberus.command.cms.CreateCmsConfigCommand;
+import com.nike.cerberus.command.consul.CreateConsulClusterCommand;
+import com.nike.cerberus.command.core.CreateBaseCommand;
+import com.nike.cerberus.command.core.UpdateStackCommand;
+import com.nike.cerberus.command.core.UploadCertFilesCommand;
+import com.nike.cerberus.command.core.WhitelistCidrForVpcAccessCommand;
+import com.nike.cerberus.command.dashboard.PublishDashboardCommand;
+import com.nike.cerberus.command.gateway.CreateCloudFrontLogProcessingLambdaConfigCommand;
+import com.nike.cerberus.command.gateway.CreateGatewayClusterCommand;
+import com.nike.cerberus.command.gateway.PublishLambdaCommand;
+import com.nike.cerberus.command.vault.CreateVaultClusterCommand;
+import com.nike.cerberus.domain.input.CerberusStack;
+import com.nike.cerberus.domain.input.Consul;
+import com.nike.cerberus.domain.input.Dashboard;
+import com.nike.cerberus.domain.input.EdgeSecurity;
+import com.nike.cerberus.domain.input.EnvironmentConfig;
+import com.nike.cerberus.domain.input.Gateway;
+import com.nike.cerberus.domain.input.ManagementService;
+import com.nike.cerberus.domain.input.Vault;
+import com.nike.cerberus.domain.input.VpcAccessWhitelist;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class EnvironmentConfigToArgsMapper {
+
+    public static final String STACK_NAME_KEY = "--stack-name";
+
+    private EnvironmentConfigToArgsMapper() {
+
+    }
+
+    public static String[] getArgs(EnvironmentConfig environmentConfig, String[] passedArgs) {
+        List<String> args = new LinkedList<>();
+        String commandName = null;
+
+        // Add all the passed args up to and including the command to run.
+        // All other args will be dropped and derived from the yaml
+        for (int i = 0; i < passedArgs.length; i++) {
+            if (StringUtils.startsWith(passedArgs[i], "-")) {

```

Comment:
This breaks if a user adds --debug, because that is not a pair.

- [ ] Fix this and add test for that